tslearn/nonmyopic.py

```diff
@@ -0,0 +1,450 @@
+                tau_star = tau
+        return tau_star + xt.shape[-1], so_far_minimum_cost
+
+    def classification(self, xt, cost_function):
+        """
+        This function classifies a single time series xt
+
+        Parameters
+        ----------------------
+        xt: vector
+            a time series that is probably incomplete but that nonetheless we want to classify
+        cost_function: funct
+            The cost function that will be use to achieve our classification purpose
+        Returns
+        -----------------------
+        int: the class which is predicted
+        int : the time of the prediction
+        float : the probability used for computing the cost
+        float : the loss when classifying
+        """
+        time_stamp = xt.shape[0]
+        xt = np.reshape(xt, (1, time_stamp))
+        time_prediction = self.minimum_time_stamp
+        stop_criterion = False
+        while stop_criterion is False:
+            minimum_tau, minimum_loss = self.minimize_integer(end_of_time=time_stamp - time_prediction,
+                                                              function_to_minimize=cost_function,
+                                                              xt=xt[:, :time_prediction])
+            if time_prediction == time_stamp:
+                result = self.classifier_[-1].predict(xt)
+                result_proba = self.classifier_[-1].predict_proba(xt)
+                stop_criterion = True
+                minimum_tau, minimum_loss = self.minimize_integer(end_of_time=time_stamp - time_prediction,
+                                                                  function_to_minimize=cost_function,
+                                                                  xt=xt[:, :time_prediction])
+                loss_exit = minimum_loss
+            elif minimum_tau == time_prediction:
+                result = self.classifier_[time_prediction - self.minimum_time_stamp].predict(xt[:, :time_prediction])
+                result_proba = self.classifier_[time_prediction -
+                                                self.minimum_time_stamp].predict_proba(xt[:, :time_prediction])
+                loss_exit = minimum_loss
+                stop_criterion = True
+            else:
+                time_prediction = time_prediction + 1
+        return result, time_prediction, result_proba, loss_exit
+
+    @staticmethod
+    def avg_prediction(time_predictions):
+
+        avgerage_time = sum(time_predictions)/len(time_predictions)
+        variance_time = np.var(time_predictions)
+        return avgerage_time, variance_time
+
+    @staticmethod
+    def accuracy_prediction(Y_test, Y_pred_test):
+
+        good_prediction = 0
+        for class_index in range(0, len(Y_test)):
+            if Y_test[class_index] == Y_pred_test[class_index]:
+                good_prediction = good_prediction + 1
+        return good_prediction / len(Y_test)
+
+    def predict(self, X_test):
```

Comment:
The scikit-learn conventions is,

```suggestion
    def predict(self, X):
```

Unless one cannot run this on other thing than the test dataset?